vignettes/demo.Rmd

```diff
@@ -0,0 +1,85 @@
+---
+title: "Use iRODS demo"
+author: "Mariana Montes"
+date: "`r Sys.Date()`"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Use iRODS demo}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+If you don't have access to an iRODS server with the [iRODS REST API](https://github.com/irods/irods_client_rest_cpp) installed,
+you can still try it out with a local demo on Linux or [WSL2](https://learn.microsoft.com/en-us/windows/wsl/install) (using Linux inside Windows).
+
+As initial requirements, you need bash, docker and the docker-compose plugin in your Linux distribution. You can check if they are installed like so (the command should provide the path to where your version is installed):
+
+```sh
+which bash
+which docker
+which docker-compose
+```
+
+You should also start a Docker daemon in your Linux session, for example:
+
+```sh
+sudo dockerd
+```
+
+If you use WSL, you can start RStudio Server with `sudo rstudio-server start` and run it from your browser (http://localhost:8787). [The Posit site](https://posit.co/download/rstudio-server/) has installation instructions for R and RStudio Server in Linux.
+
+Then, in your script or console, load `{rirods}` and run `use_irods_demo()`:
+
+```{r}
+library(rirods)
+use_irods_demo()
+```
+
+The first time you run it it may ask you questions to guiding through generating the first docker image. Once you get the message shown above, you can create the configuration file (if it doesn't exist yet) and authenticate. Here we authenticate with the default "rods" account, but you could have another account as well; remember that running `iauth()` will open an interactive dialog to provide a username and password.
+
+```{r}
+create_irods("http://localhost/irods-rest/0.9.3", "/tempZone/home", overwrite = TRUE)
+iauth("rods", "rods")
+ils()
+```
+
+If you have any issues running the demo, check:
+
+- that you are working from Linux (or WSL2)
+- that you have `docker`, and `docker-compose` installed
+- that you have (re-)installed the package locally _after_ checking all these to be true (and restarted R)
+- that you use the same username and password in `use_irods_demo()` and in `iauth()`.
```

Comment:
Maybe also check whether you need `sudo`. https://docs.docker.com/engine/install/linux-postinstall/

Reply:
Good point, adding yourself to the `docker` group...